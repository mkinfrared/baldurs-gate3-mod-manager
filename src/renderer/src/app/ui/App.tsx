import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { MemoryRouter as Router } from "react-router-dom";

import { AppRoutes } from "@renderer/routes";
import { trpc, trpcClient } from "@renderer/shared/lib/helpers";
import { AppNavBar } from "@renderer/widgets/ui";

import css from "./App.module.scss";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      suspense: true,
    },
  },
});

/**
 * https://trpc.io/docs/v9/react
 * https://github.com/awohletz/electron-prisma-trpc-example/blob/main/src/client/renderer.ts
 */

const App = () => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>
      <Router>
        <div className={css.App} data-testid="App">
          <aside>
            <AppNavBar className={css.navBar} />
          </aside>
          <main>
            <AppRoutes />
          </main>
        </div>
      </Router>
    </QueryClientProvider>
  </trpc.Provider>
);

export { App };

export default App;
