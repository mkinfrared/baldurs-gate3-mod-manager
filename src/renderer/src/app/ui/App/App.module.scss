.App {
  display: grid;
  grid-template-columns: auto 1fr;
  min-height: 100dvh;

  main {
    display: grid;
    gap: var(--spacing-10);
    grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
    padding: var(--spacing-10);
  }
}
