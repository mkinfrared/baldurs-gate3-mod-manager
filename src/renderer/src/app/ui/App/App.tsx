import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { MemoryRouter as Router } from "react-router-dom";

import { AppRoutes } from "@renderer/routes";
import { trpc, trpcClient } from "@renderer/shared/lib/helpers";
import { AppNavBar } from "@renderer/widgets/ui";

import css from "./App.module.scss";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      suspense: true,
    },
  },
});

const App = () => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>
      <Router>
        <div className={css.App}>
          <aside>
            <AppNavBar />
          </aside>
          <main>
            <AppRoutes />
          </main>
        </div>
      </Router>
    </QueryClientProvider>
  </trpc.Provider>
);

export { App };

export default App;
