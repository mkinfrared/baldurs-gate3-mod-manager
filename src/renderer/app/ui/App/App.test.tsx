import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import { App } from "./App";

jest.mock("renderer/routes");

describe("App", () => {
  it("should render", () => {
    expect(render(<App />)).toBeTruthy();
  });

  it("should match the snapshot", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
