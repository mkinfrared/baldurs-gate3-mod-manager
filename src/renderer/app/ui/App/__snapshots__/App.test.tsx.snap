// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <aside>
      <div
        class="AppNavBar"
        data-testid="AppNavBar"
      >
        AppNavBar works!!
      </div>
    </aside>
    <main />
  </div>
</div>
`;
